spec/reek/smells/nested_iterators_spec.rb

```diff
@@ -271,6 +271,15 @@ def bad(fred)
       expect(src).to reek_of(:NestedIterators, count: 2).with_config(config)
     end
   end
+
+  it 'should analyze safe-navigation-operator without error' do
+    src = '
+      def f
+        x&.tap {|_| }
+      end
+    '
+    expect(src).not_to reek_of(:NestedIterators, count: 2)
```

Comment:
I'd prefer to have some actual tests for NestedIterators with safe navigation that show it detecting actual smells. I vaguely recall having some of those tests lying around from earlier when making them pass was blocked by an issue in the parser gem.

TL;DR: I'm ok with merging this one without a test for NestedIterators right now (so basically removing this these lines from the PR), and then I will add a set of tests this weekend.